### PR TITLE
Exclude columns

### PR DESCRIPTION
**Pull request by iamjem**
Provide a mechanism for excluding table columns

### CODE REVIEW

File: analyticscloud/importers/db.py [hunk 1]
```diff
@@ -101,9 +101,10 @@ def metadata_dict(dburl, table, extended=None, schema='public'):
 
     metadata, fields = metadata_factory(table.name)
     metadata['objects'][0].update(extended)
-    for col in table.columns:
-        api_name = col.name
-        display_name = col.name
+
+    columns = table.columns if excludes is None else exclude_columns(table.columns, excludes)
```
Comment (sibson) [inline analyticscloud/importers/db.py]: Could we move the excludes=None logic into exclude_columns? seems like it should handle that case itself and would make the callee cleaner.

Comment (iamjem) [inline analyticscloud/importers/db.py]: Sure, I thought about that initially too, it does eliminate some boilerplate if/else logic

**Comment by iamjem** [comment]
@sibson feels a little weird having a function for filtering columns which can accept no filtering criteria, but it does remove a line or two of code form the calling methods.

**Comment by sibson** [comment]
I'd say it makes the function more robust but maybe not add the kwargs default exclude=None.  It should require the arg but handle if its None.

+1

**Comment by iamjem** [comment]
I thought the default was appropriate since its an optional argument. Otherwise consumers of the function would have to look at the function body to understand what null value they should be passing. Either way, an insignificant quibble, and I'm fine having it exist either way.

**Pull request merged**